config/routes.rb

```diff
@@ -0,0 +1,7 @@
+Rails.application.routes.draw do
+  get '/users/:user_id/posts', to: 'posts#index', as: 'user_posts'
+  get '/users/:user_id/posts/id', to: 'posts#show', as: 'user_post'
+  get '/users', to: 'users#index', as: 'users'
+  get '/users/id', to: 'users#show', as: 'user'
+root 'users#index'
+end
```

Comment:
## Optional
- [ ] I highly suggest using Resource routing allows you to quickly declare all of the common routes for a given resourceful controller
you can simply do 
` resources :users, only: [:index, :show] do
    resources :posts, only: [:index, :show]
  end`

Reply:
Thank you @DeliceLydia 😊